src/clarity-angular/tabs/vtabs.ts

```diff
@@ -0,0 +1,212 @@
+     * 4. Sets the first link and first content as active, if needed.
+     *
+     * No params, called as part of the angular lifecycle
+     */
+    ngAfterContentInit() {
+        // Initialize tabs and content.
+
+        this.vtabLinks.forEach((link, index) => {
+            link.id = "clr-vtab-link-" + nbVTabLinks.toString();
+            nbVTabLinks++;
+            this.mapContentId(index, link.id);
+            // subscribe to the links click event
+            link.activatedChange.subscribe((updatedLink: VTabsNavLink) => {
+                this.updateForActiveLink(updatedLink);
+            });
+
+            //check for activated link
+            if (link.activated) {
+                this.activatedLink = index;
+            }
+        });
+
+        // Activate the first tab and content if nothing has been set with the VTabsNavLink.activated @Input().
+        if (!this.activatedLink) {
+            let testLink = this.vtabLinks.first;
+            testLink.activated = true;
+
+            let testContent: VTabsContent = this.vtabContents.first;
+            testContent.activated = true;
+        } else {
+            this.activateTabContentAtIndex(this.activatedLink);
+        }
+    }
+
+    /**
+     * @function updateForActiveLink
+     *
+     * @description
+     * Method to update all the VTabNavLinks when an activatedChange event is heard by:
+     *
+     * 1. Deactivating all the linkItem.ids that don't match the updatedLink passed as arg
+     * 2. Identifying the index of the newly activated tab link so we can activate corresponding VTabContent
+     * 3. Activating the corresponding VTabsContent item.
+     *
+     * @param updatedLink -  a VTabNavLink that was clicked on.
+     *
+     */
+    updateForActiveLink(updatedLink: VTabsNavLink) {
+        let linkIndex: number; // Map between link and content
+        this.vtabLinks.forEach((linkItem: VTabsNavLink, index: number) => {
+            if (linkItem.id !== updatedLink.id) {
+                linkItem.activated = false;
+            } else {
+                linkIndex = index; // Position of the link that was clicked.
+                this.activatedLink = index;
+            }
+        });
+
+        // Update content for index
+        if (typeof linkIndex !== "undefined") { // Remember, remember 0 is also false.
+            this.activateTabContentAtIndex(linkIndex);
+        }
+
+        // Broadcast the link change event
+        this.linkChange.emit(updatedLink);
+    }
+
+    /**
+     * @function activateTabContentAtIndex
+     *
+     * @description
+     * Method to update VTabContent items as follows:
+     *
+     * 1. Deactivate all VTabContent items not equal to the linkIndex arg
+     * 2. Activate the VTabContent iotem that has the same index as the linkIndex arg
+     *
+     * @param linkIndex - the index to activate in this.vtabsContent QueryList
+     */
+    activateTabContentAtIndex(linkIndex: number) {
```

Comment:
I generally make very small unit tests that test a specific path. I would have individual tests that validated the `contentIndex === linkIndex` as well as `content.activated = false` fallthrough.

@youdz also directed me to divide my tests into TypescriptAPI, Template API, and View/Behavior groupings.

What you've written is a high-level test that goes through both states that would fit in the Template API grouping. But you would also want to verify any CSS classes are added to the component as expected in a test in the View/Behavior grouping.

An assumption that I see here is that vtabContents is always populated. I would hope to see more resiliency in handling an edge case if vtabContents was an empty array as well as if it came through as undefined or null.

These are questions I've been pressed to answer in the wizard. Do we throw an error if we find out vtabContents has been cleared out? Or something in the architecture messes up?

Your tests are good. And bonus points for using @youdz text context helpers. My issue is more around the rigor being expected of the tests I'm having to write is not reflected here.

So I'm wondering (for my own self) why there is a delta?

Your one test (were it in the wizard) would be broken up into 3-4 tests across 2-3 groupings. Yet I don't see any groupings here and the tests are much broader/birds-eye than I've been told to write.

So I don't know what to tell you other than this is not how I have been told to write tests for components. I don't know why I've been told to do them differently.